parallel slopes coefficient interpretation

katyshea opened the pull request.

katyshea committed:
parallel slopes coefficient interpretation. Check case expression formatting

katyshea committed:
interaction model and comparison table

katyshea commented:
Quarto preview not working. I don't understand the errors (there are several)

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -406,6 +406,121 @@ That's why the lines are parallel.
+Or we can plug in the definition of slope:
+
+$\beta_A$ is the difference in the mean value of birthweight that we would expect, for a one-unit difference in gestational age, adjusting for sex.
+$$\beta_A &= E[Y|m,a+1] - E[Y|m,a]$$
```
I think the `&=` alignment notation doesn't work outside of `aligned` contexts
```suggestion
$$\beta_A = E[Y|m,a+1] - E[Y|m,a]$$
```

d-morrison replied:
@katyshea I think that `&` is the cause of this error in the quarto preview workflow:

![image](https://github.com/user-attachments/assets/30658a94-6e6d-4233-b010-f41f2003a3ff)

katyshea replied:
Oh, yup. Must've copied and pasted without paying attention

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -406,6 +406,121 @@ That's why the lines are parallel.
+
+Or we can plug in the definition of slope:
+
+$\beta_A$ is the difference in the mean value of birthweight that we would expect, for a one-unit difference in gestational age, adjusting for sex.
```
```suggestion
$\beta_A$ is the difference in mean birthweight per one-unit difference in gestational age, adjusting for sex.
```

katyshea committed:
Update Linear-models-overview.qmd

beta A interpretation update

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea committed:
Update Linear-models-overview.qmd

remove unnecessary &

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea committed:
fix open bracket

katyshea committed:
remove returns on last line of aligned environment

katyshea committed:
fix formatting issues

katyshea committed:
more formatting

katyshea committed:
formatting

katyshea committed:
make interaction coefficient solution smaller

katyshea committed:
make interaction coefficient solution smaller

katyshea committed:
render issue

katyshea committed:
realign interaction equations

katyshea committed:
remove smaller

katyshea committed:
slightly improved table formatting

katyshea committed:
try LaTeX table

katyshea committed:
fix table formatting

katyshea committed:
add section header

katyshea committed:
add section header

katyshea committed:
merge conflict

Merge commit '3e0cedbed1f85475119027a7914b96a1223ce477'

#Conflicts:
#	Linear-models-overview.qmd

d-morrison committed:
Merge branch 'main' into linreg-coef-interpretation

d-morrison committed:
Merge branch 'main' into linreg-coef-interpretation

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -407,6 +407,130 @@ That's why the lines are parallel.
+:::{#exr-interpret-intercept}
+$\beta_0 = ?$
+:::
+
```
let's put slide breaks between exercises/theorems and solutions/proofs:

```suggestion

---

```

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -407,6 +407,130 @@ That's why the lines are parallel.
+:::
+
+:::{.solution}
+$\beta_0$ is the mean birthweight for a female with gestational age 0 weeks.
+
+$$
+\begin{aligned}
+E[Y|0,0] &= \beta_0 + \beta_M 0 + \beta_A 0\\
+&= \beta_0
+\end{aligned}
+$$
```
rearrange to derivation then conclusion?

```suggestion
$$
\begin{aligned}
\E{Y|0,0} &= \beta_0 + \beta_M 0 + \beta_A 0\\
&= \beta_0
\\ \beta_0 &= \E{Y|0,0}
\end{aligned}
$$

::: notes
$\beta_0$ is the mean birthweight for a female with gestational age 0 weeks.
:::

```

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+:::
+
+::: {.solution}
+$\beta_{AM}$ is the difference in slope of mean birthweight with respect to gestational age between males and females.
```
as above; move conclusion to end of solution? I could be convinced the current ordering is better; we can discuss when we meet tomorrow

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+
+Slope among males:
+$$
+\begin{aligned}
+ E[Y|a+1,1] - E[Y|a,1] = &\beta_0 + \beta_M 1 + \beta_A(a+1) + \beta_{AM}1(a+1)\\ 
+  &- (\beta_0 + \beta_M 1 + \beta_A(a) + \beta_{AM}1(a))\\
+ = &\beta_A + \beta_{AM}
+ \end{aligned}
+$$
+
+Slope among females:
+$$
+\begin{aligned}
+ E[Y|a+1,0] - E[Y|a,0] = &\beta_0 + \beta_M0 + \beta_A(a+1) + \beta_{AM}0(a+1) \\
+ &- (\beta_0 + \beta_M 0 + \beta_A(a) + \beta_{AM}0(a))\\
+ = &\beta_0 + \beta_A(a+1) - (\beta_0 + \beta_A(a))\\
+ = &\beta_A 
+\end{aligned}
+$$
+
+Difference in slopes:
+$$
+\begin{aligned}
+ (E[Y|a+1,1] - E[Y|a,1]) - (E[Y|a+1,0] - E[Y|a,0]) &= \beta_A + \beta_{AM} - \beta_A\\
+ &= \beta_{AM}
+\end{aligned}
```
switch to, or add, derivative version?

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+
+#### Coefficient Interpretation
+
+
```
add exercise for $\beta_M$ in interaction model?

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -407,6 +407,130 @@ That's why the lines are parallel.
+
+Or we can plug in the definition of slope:
+
+$\beta_A$ is the difference in mean birthweight per one-unit difference in gestational age, adjusting for sex.
```
end speaker note here?

```suggestion
$\beta_A$ is the difference in mean birthweight per one-unit difference in gestational age, adjusting for sex:
:::
```

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -407,6 +407,130 @@ That's why the lines are parallel.
+$$\beta_A = E[Y|m,a+1] - E[Y|m,a]$$
+:::
+
+---
+
+:::{.notes}
+Exchangeability and consistency have not been assessed; so we are not discussing potential outcomes (causality), only observed outcomes.
+:::
+
+---
```
this is currently another notes-only slide; merge with previous and keep this comment as a speaker note?

```suggestion

:::{.notes}
Exchangeability and consistency have not been assessed; so we are not discussing potential outcomes (causality), only observed outcomes.
:::

---
```

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+
+#### Compare coefficient interpretations
+
+:::{.notes .smaller}
```
```suggestion
:::{#tbl-compare-coef-interps}
```

- table goes on slide, not just in speaker notes
- I really wish the `.smaller` tag worked on tables, but I don't think it does; seems to be an unimplemented feature: https://github.com/quarto-dev/quarto-cli/issues/1153

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+  $\beta_A$     | $E[Y\mid a+1,m] - E[Y\mid a,m]$    | $E[Y\mid a+1,0] - E[Y\mid a,0]$
+  $\beta_M$     | $E[Y\mid a,1] - E[Y\mid a,0]$      | $E[Y\mid A=0,1] - E[Y\mid A=0,0]$
+  $\beta_{AM}$  |                                    | $\begin{aligned}(&E[Y\mid a+1,1] - E[Y\mid a,1])\\ &- (E[Y\mid a+1,0] - E[Y\mid a,0])\end{aligned}$
+:::
```
```suggestion

Coefficient interpretations, by model structure

:::
```

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+#### Compare coefficient interpretations
+
+:::{.notes .smaller}
+  $\mu(a,s)$    | $\beta_0 + \beta_M m + \beta_A a$ | $\beta_0 + \beta_M m + \beta_A a + \beta_{AM} ma$
```
I created some color macros that we can use to highlight important parts of equations:

```suggestion
  $\mu(a,s)$    | $\beta_0 + \beta_M m + \beta_A a$ | $\beta_0 + \beta_M m + \beta_A a + \red{\beta_{AM} ma}$
```

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -625,6 +749,57 @@ That's how the lines in the graph ended up non-parallel.
+  $\beta_0$     | $E[Y\mid a,1] - E[Y\mid a,0]$      | $E[Y\mid A=0,1] - E[Y\mid A=0,0]$
+  $\beta_A$     | $E[Y\mid a+1,m] - E[Y\mid a,m]$    | $E[Y\mid a+1,0] - E[Y\mid a,0]$
+  $\beta_M$     | $E[Y\mid a,1] - E[Y\mid a,0]$      | $E[Y\mid A=0,1] - E[Y\mid A=0,0]$
+  $\beta_{AM}$  |                                    | $\begin{aligned}(&E[Y\mid a+1,1] - E[Y\mid a,1])\\ &- (E[Y\mid a+1,0] - E[Y\mid a,0])\end{aligned}$
```
switch to difference of slopes (derivatives) version?

d-morrison reviewed (changes requested):
Thanks for typing this up! I left some comments (it's easiest to read them in-context on the [Files changed](https://github.com/d-morrison/rme/pull/54/files) tab); glad to discuss during our meeting, or @ me here with questions.

d-morrison committed:
Merge branch 'main' into linreg-coef-interpretation

katyshea committed:
Update Linear-models-overview.qmd

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea commented on Linear-models-overview.qmd:
```diff
@@ -407,6 +407,130 @@ That's why the lines are parallel.
+---
+
+#### Coefficient Interpretation
+:::{.notes}
```
Not what I intended. I forgot about the slides completely

katyshea committed:
Update Linear-models-overview.qmd

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea committed:
formatting for revealjs and restructuring

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea commented on Linear-models-overview.qmd:
```diff
@@ -407,6 +407,135 @@ That's why the lines are parallel.
+
+---
+
+More precisely written:
+$$ 
+\E{Y|A=a,M=m} = \mu(a,m) = \bigg\{
+\begin{matrix} 
+\beta_0 + \beta_M m +\beta_A a, & \text{for } m\in \set{0,1} \\ 
+\text{undefined}, & \text{for } m \notin \set{0,1}
+\end{matrix}
+$$
+The model is undefined for $m \notin \set{0,1}$, so the derivative with respect to $m$ doesn't exist.
```
Move to beginning of section?

d-morrison replied:
> Move to beginning of section?

I think it belongs at the beginning of the solution below (in a `notes` div); it explains why we can't take the derivative to find $\beta_M$

katyshea committed:
adding in derivative versions and reorganizing solutions

katyshea committed:
Merge branch 'main' into linreg-coef-interpretation

katyshea commented:
@d-morrison Quarto preview is failing. I don't understand the error. It knits just fine from R Studio

katyshea committed:
try to fix quarto preview

katyshea committed:
Merge branch 'linreg-coef-interpretation' of https://github.com/d-morrison/rme into linreg-coef-interpretation
updating local branch

katyshea committed:
space after backslash

d-morrison committed:
Merge branch 'main' into linreg-coef-interpretation

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -412,6 +412,133 @@ That's why the lines are parallel.
+
+---
+
+:::{#thm-beta-m-interpretation}
+$$ \beta_M = E[Y|a,1] - E[Y|a,0] $$
+In words: $\beta_M$ is the difference in mean birthweight between males and females adjusting for age.
+:::
+
+:::{.proof}
+$$
+\begin{aligned}
+E[Y|a,1] &= \beta_0 + \beta_M 1 + \beta_A a \\
+& = \beta_0 + \beta_M + \beta_A a \\
+E[Y|a,0] &= \beta_0 + \beta_M 0 + \beta_A a \\
+& = \beta_0 + \beta_A a
+\end{aligned}
+$$
+So: 
+$$
+\begin{aligned}
+E[Y|a,1] - E[Y|a,0] &= (\beta_0 + \beta_M + \beta_A a) - (\beta_0 + \beta_A a) \\
+&= \beta_M
+\end{aligned}
+$$
+:::
```
convert to exercise + solution?

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -630,6 +757,109 @@ That's how the lines in the graph ended up non-parallel.
+::: {#exr-male-coef-intxn-model}
+What is the interpretation of $\beta_{M}$?
+:::
+
```
```suggestion

---

```
please add slide breaks before each exercise and between each exercise and corresponding solution, so that I can put the exercises up on the screen by themselves, and the solutions are on the following slide

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -630,6 +757,109 @@ That's how the lines in the graph ended up non-parallel.
+::: {#exr-interaction-coef}
+What is the interpretation of $\beta_{AM}$?
+:::
+
```
```suggestion

---

```

d-morrison reviewed (changes requested):
almost there! a few more changes requested

katyshea commented on Linear-models-overview.qmd:
```diff
@@ -412,6 +412,133 @@ That's why the lines are parallel.
+
+---
+
+More precisely written:
+$$ 
+\E{Y|A=a,M=m} = \mu(a,m) = \bigg\{
+\begin{matrix} 
+\beta_0 + \beta_M m +\beta_A a, & \text{for } m\in \set{0,1} \\ 
+\text{undefined}, & \text{for } m \notin \set{0,1}
+\end{matrix}
+$$
+The model is undefined for $m \notin \set{0,1}$, so the derivative with respect to $m$ doesn't exist.
```
```suggestion
:::{.notes}
More precisely written:
$$ 
\E{Y|A=a,M=m} = \mu(a,m) = \bigg\{
\begin{matrix} 
\beta_0 + \beta_M m +\beta_A a, & \text{for } m\in \set{0,1} \\ 
\text{undefined}, & \text{for } m \notin \set{0,1}
\end{matrix}
$$
The model is undefined for $m \notin \set{0,1}$, so the derivative with respect to $m$ doesn't exist.
:::
```

katyshea committed:
add slide breaks from code review

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea committed:
moving more precisely written... section

d-morrison reviewed (changes requested):
almost there - please see comments, and look through the [preview](https://d-morrison.github.io/rme/pr-preview/pr-54/Linear-models-overview.html) to make sure these exercises look correct in render?

katyshea committed:
fix exercise appearance

katyshea committed:
add slide break

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

katyshea committed:
slidebreak before exr-beta-m-interpretation not adding horizontal line

katyshea committed:
fix slide break merge conflict

Merge commit 'ca893d4c80f7d984795421fef367442e177cb06e'

#Conflicts:
#	Linear-models-overview.qmd

katyshea committed:
formatting book and slides

katyshea committed:
consistent order of M and A in mu function and expectation notations

d-morrison committed:
Merge branch 'main' into linreg-coef-interpretation

d-morrison commented:
(@katyshea I just merged in main accidentally; forgot which PR I was looking at)

d-morrison commented:
> (@katyshea I just merged in main accidentally; forgot which PR I was looking at)

let me know if that breaks anything on your end, glad to help fix or undo

katyshea committed:
one more check for typos in variable notation in Exp and mu funciton

d-morrison commented on Linear-models-overview.qmd:
```diff
@@ -732,7 +732,7 @@ pred_male <-
 $$
 \ba
-E[Y|A = 0, X_2 = 36]
+E[Y|M = 1, X_2 = 36]
```
are these $X_2$s supposed to be $A$ (age)? (I realize this section wasn't originally part of this PR; just noticed as I reviewed your changes)

katyshea replied:
Yes, I think so. I just noticed when I was checking through my notation that this one had an A in the wrong spot, so I changed it. I didn't change the X2 because I wasn't sure if that was a different notation on purpose or not

d-morrison replied:
ok, I logged this as a separate issue (#91); we can deal with it later (or next year...)

d-morrison committed:
Merge branch 'main' into linreg-coef-interpretation

d-morrison reviewed (approved):
looks good - thanks!
